### PR TITLE
Establishes an early version of ELO Modeling for proficiency.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Related documentation (not publicly accessible): https://docs.google.com/document/d/1e6m1edRSBnom-Pn5W6E4EkL2mB_mFITaO3S-y-CmhU0

### CODE REVIEW

**Comment by ashercodeorg** [comment]
CC @ryansloan

File: experimental/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Experimental
+
+This directory is intended to contain files that should be version-controlled,
+are experimental in nature, and do not have (long term) dependencies from
+production.
+
+In particular, files associated with experiments running in production do not
+belong here.
```
Comment (Hamms) [inline experimental/README.md]: Oh wow, I love this idea

File: experimental/csf_proficiency/elo_modeling.rb [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  for_loops: 4,
+  events: 5,
+  variables: 6,
+  functions: 7,
+  functions_with_params: 8,
+  conditionals: 9
+}.freeze
+
+# @param file [String] filepath to a CSV containing UserLevel summary data.
+#   Each for of the CSV should contain a user_id, level_id, best_result,
```
Comment (Hamms) [inline experimental/csf_proficiency/elo_modeling.rb]: typo: s/for/line

File: experimental/csf_proficiency/elo_modeling.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+# @param best_result [Integer] the best result of the UserLevel.
+# @param attempts [Integer] the number of attempts of the UserLevel.
+# @param num_hints [Integer] the number of hints associated with the UserLevel.
+# @param num_authored_hints [Integer] the number of authored hints associated
+#   with the UserLevel.
+# @return [Float] whether the level was passed perfectly without hints.
+def perfect_without_hints(best_result, attempts, num_hints, num_authored_hints)
+  raise ArgumentError.new('FreePlay') if best_result == 30
+  return 1.0 if best_result == 100 && num_hints == 0 && num_authored_hints == 0
```
Comment (Hamms) [inline experimental/csf_proficiency/elo_modeling.rb]: I thought we had decided to start using `> 30` in order to accommodate diagnostic results?

**Comment by ashercodeorg** [comment]
It has been run on some production data (see the [updated] PR summary for a document link).

That said, the script is far from polished (e.g., it is possible to generate a `NoMethodError` exception). As I'm trying to get this in the hands of ryansloan (also anyone else that wants to play) as quickly as possible, I'm taking the launch and iterate approach.

File: experimental/README.md (hunk 1 above)
Comment (ashercodeorg) [reply experimental/README.md]: Borrowing the idea from a former employee. My intent is to submit the PR establishing the directory, merge staging, and eliminate the directory and readme from this PR.

File: experimental/csf_proficiency/elo_modeling.rb (hunk 3 above)
Comment (ashercodeorg) [reply experimental/csf_proficiency/elo_modeling.rb]: Done.

File: experimental/csf_proficiency/elo_modeling.rb (hunk 2 above)
Comment (ashercodeorg) [reply experimental/csf_proficiency/elo_modeling.rb]: Done.

**Pull request merged**